src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  const obj = { ...state };
+
+  for (const element of actions) {
+    const { type, extraData, keysToRemove } = element;
+
+    switch (type) {
+      case 'addProperties': {
+        for (const key in extraData) {
+          obj[key] = extraData[key];
+        }
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of keysToRemove) {
+          if (obj[key]) {
```

Comment:
```suggestion

```
There is no need to provide an additional check here, the delete operator wouldn't throw an error if the property doesn't exist